File: src/dataset/javascript/sieve_exercism-ThomasZumsteg/gemini/GeminiFlash_sieve_v4.js
var Sieve = function(limit) {
  this.primes = [];
  const isPrime = new Array(limit + 1).fill(true);
  isPrime[0] = isPrime[1] = false;

  for (let number = 2; number <= Math.sqrt(limit); number++) {
    if (isPrime[number]) {
      for (let multiple = number * number; multiple <= limit; multiple += number) {
        isPrime[multiple] = false;
      }
    }
  }

  for (let number = 2; number <= limit; number++) {
    if (isPrime[number]) {
      this.primes.push(number);
    }
  }
};

module.exports = Sieve;

File: src/dataset/python/sieve_exercism-ThomasZumsteg/gemini/GeminiFlash_sieve_v4.py
"""Finds primes numbers"""

def sieve(lim):
    """Implimentation of sieve of Eratosthenes"""
    if lim < 2:
        return []

    primes = []
    is_prime = [True] * (lim + 1)
    is_prime[0] = is_prime[1] = False

    for p in range(2, int(lim**0.5) + 1):
        if is_prime[p]:
            for i in range(p*p, lim + 1, p):
                is_prime[i] = False

    for p in range(2, lim + 1):
        if is_prime[p]:
            primes.append(p)
    return primes